src/main/resources/js/providedLibs.js

```diff
@@ -10,26 +10,31 @@
 
     const getClassName = path => path.substring(path.lastIndexOf('.') + 1)
 
-    function getClass(path) {
+    function getClass(path, failSilently = false) {
         const clazz = Java.type(path);
         if (clazz.class instanceof Class)
             return clazz;
-        throw new Error(`moduleProvidedLibs: Could not load class "${path}"`);
+        if (!failSilently)
+            throw new Error(`moduleProvidedLibs: Could not load class "${path}"`);
     }
 
     function loadClass(path, className = getClassName(path)) {
         global[className] = getClass(path);
     }
 
+    function loadPossibleClass(path, className = getClassName(path)) {
```

Comment:
The functionality in `providedLibs.js` should be the exact same across versions to discourage version-specific user code, so I'm not a fan of this. If a class doesn't exist in all versions, we should make a wrapper for it. Since this PR is already huge, I'd suggest reverting these changes and just commenting out the `loadClass` calls which can fail, and we can address it in the future.